src/vellum/workflows/expressions/does_not_contain.py

```diff
@@ -24,7 +25,9 @@ def resolve(self, state: "BaseState") -> bool:
         # https://app.shortcut.com/vellum/story/4658
         lhs = resolve_value(self._lhs, state)
         if not isinstance(lhs, (list, tuple, set, dict, str)):
-            raise ValueError(f"Expected a LHS that supported contains, got: {lhs.__class__.__name__}")
```

Comment:
Going to fast follow this PR to replace all other `ValueError`s in expressions with this new exception